[ATB-140] Open source guidelines (#68)
HewlettPackard/galadriel: doc/USAGE.md

| Flag      |  Type  | Description               |
|-----------|--------|---------------------------|
| `-t`      | string | SPIRE server trust domain |

### `galadriel-server create relationship`
| Flag      |  Type  | Description                 |
|:----------|:-------|:----------------------------|
| `-a`      | string | SPIRE Server trust domain A |
| `-b`      | string | SPIRE Server trust domain A |

### `galadriel-server generate token`
| Flag      |  Type  | Description               |
|-----------|--------|---------------------------|
| `-t`      | string | SPIRE server trust domain |

### `galadriel-server list`
| Command         | Description                                           |
|-----------------|-------------------------------------------------------|
| `members`       | List all members stored in the Galadriel Server       |
| `relationships` | List all relationships stored in the Galadriel Server |

# Galadriel Harvester CLI
The Galadriel Harvester CLI contains the functionality to run the Galadriel Harvester while attaching it to the Galadriel Server instance, based on the token used as a argument:

### `galadriel-harvester run`

> | Flag      |  Type  | Description               |
> |-----------|--------|---------------------------|
> | `-t`      | string | SPIRE server trust domain |

Reviewer: ```suggestion
| Flag      |  Type  | Description               | Default                 |
|-----------|--------|---------------------------| ----------------------- |
| `-c`      | string | Config file path          | conf/server/server.conf |
```
nit: I wouldn't worry much about table formatting in Markdown, something like this works and it's easier to maintain (changes don't require to reformat the whole table)
```
| Col1 | Col2
| -- | --
| Something | Something else
```

Reviewer: sorry I meant to add a new flag, not to suggest a change

# Galadriel Server Configuration File

> You can find the Galadriel Server configuration file at `galadriel/conf/server`

Reviewer: ```suggestion
You can find the default Galadriel Server configuration file at `conf/server/server.conf`
```

| Configuration    |  Description                                     | Default                         |
|------------------|--------------------------------------------------|---------------------------------|
| `listen_address` | IP address or DNS name of the Galadriel server.  |  localhost                      |
| `listen_port`    | HTTP Port number of the Galadriel server.        |  8085                           |
| `socket_path`    | Path to bind the Galadriel Server API socket to. |  /tmp/galadriel-server/api.sock |

# Galadriel Harvester Configuration File

> You can find the Galadriel Server configuration file at `galadriel/conf/harvester`

Reviewer: ```suggestion
You can find the default Galadriel Harvester configuration file at `conf/harvester/harvester.conf`
```

# Galadriel Server CLI
The Galadriel Server CLI contains the functionality to:
### `galadriel-server create member`
| Flag | Type | Description |
|--|--|--|

> | `-t, --trustDomain`| string | SPIRE server trust domain |

Reviewer: super nit:
```suggestion
| `-t`, `--trustDomain`| string | SPIRE server trust domain |
```
and in the following flags too

Author: Nice catch

The Galadriel Harvester CLI contains the functionality to run the Galadriel Harvester while attaching it to the Galadriel Server instance, based on the token used as a argument:

### `galadriel-harvester run`
| Flag | Type | Description |
|--|--|--|
| `-t, --token` | string | SPIRE server trust domain |
| `-c, --config` | string | Config file path |

# Galadriel Server Configuration File
You can find the default Galadriel Server configuration file at `conf/server/server.conf`

| Configuration | Description | Default |
|--|--|--|
| `listen_address` | IP address or DNS name of the Galadriel server. | localhost |
| `listen_port` | HTTP Port number of the Galadriel server. | 8085 |
| `socket_path` | Path to bind the Galadriel Server API socket to. | /tmp/galadriel-server/api.sock |

# Galadriel Harvester Configuration File
You can find the default Galadriel Harvester configuration file at `conf/harvester/harvester.conf`

| Configuration | Description | Default
|--|--|--|
| `spire_socket_path` | SPIRE Server Socket of the instance to manage. | /tmp/spire-server/private/api.sock |
| `server_address` | Upstream Galadriel Server DNS name or IP address with port. | localhost:8085 |

> | `bundle_updates_interval` | Sets how often to check for bundle rotation. | 5s |

Reviewer: Just to clarify a little bit about properties, required, default values, and config files.

A property has either a default value or is required. If you don't set a property and it's required, galadriel run will fail. If the property has a default value it will take this value instead.
You can identify the properties with a default value by the `Default: XXX` comment in the config file. If it doesn't has this value, it means is required.

Additionally, all properties have a value in the config files, despite they having a default value or being required. This is to facilitate users to quickly clone and run both servers and harvester without needing to change anything.

I believe we should document whether a property is required or has a default value, not what we have in the config files for convenience. Maybe something like this

| Configuration | Description | Required | Default
|--|--|--|--|
| `spire_socket_path` | SPIRE Server Socket of the instance to manage. | | `/tmp/spire-server/private/api.sock`
| `server_address` | Upstream Galadriel Server DNS name or IP address with port. | Yes |
| `bundle_updates_interval` | Sets how often to check for bundle rotation. | | `30s`
| `log_level` | Application log level. One of: `TRACE`, `DEBUG`, `INFO`, `WARN`, `ERROR`, `FATAL`, `PANIC` | | `INFO`

--

Another topic...

The default we have for `bundle_updates_interval` is `30s` but we have `5s` set as an example. This is a bit confusing... how about changing the value set in `harvester.conf` to `30s` to match the default value?
```suggestion
| `bundle_updates_interval` | Sets how often to check for bundle rotation. | | 30s
```

Reviewer: We could also clarfiy in the config files as a comment whether a property is required or not.

# Galadriel Server CLI
The Galadriel Server CLI contains the functionality to:
### `galadriel-server create member`
| Flag | Type | Required | Description |
|--|--|--|--|
| `-t`, `--trustDomain`| string | Yes | SPIRE server trust domain |

### `galadriel-server create relationship`
| Flag | Type | Required | Description |
|--|--|--|--|
| `-a`, `--trustDomainA` | string | Yes | SPIRE Server trust domain A |
| `-b`, `--trustDomainB` | string | Yes | SPIRE Server trust domain B |

### `galadriel-server generate token`
| Flag | Type | Required | Description |
|--|--|--|--|
| `-t`, `--trustDomain` | string | Yes | SPIRE server trust domain |

### `galadriel-server list`
| Command | Description |
|--|--|
| `members` | List all members stored in the Galadriel Server |
| `relationships` | List all relationships stored in the Galadriel Server |

# Galadriel Harvester CLI
The Galadriel Harvester CLI contains the functionality to run the Galadriel Harvester while attaching it to the Galadriel Server instance, based on the token used as a argument:

### `galadriel-harvester run`
| Flag | Type | Required | Description |
|--|--|--|--|

> | `-t`, `--token` | string | Yes | SPIRE server trust domain |

Reviewer: ```suggestion
| `-t`, `--token` | string | Yes | Token generated by the Galadriel Server for this trust domain |
```

Reviewer: seems like you change the description in the other command instead 😄

| `-c`, `--config` | string |  | Config file path. If not set uses the default value: `conf/harvester/harvester.conf` |

# Galadriel Server Configuration File
You can find the default Galadriel Server configuration file at `conf/server/server.conf`

| Configuration | Description | Default |
|--|--|--|
| `listen_address` | IP address or DNS name of the Galadriel server. | localhost |
| `listen_port` | HTTP Port number of the Galadriel server. | 8085 |
| `socket_path` | Path to bind the Galadriel Server API socket to. | /tmp/galadriel-server/api.sock |
| `log_level` | Application log level. One of: `TRACE`, `DEBUG`, `INFO`, `WARN`, `ERROR`, `FATAL`, `PANIC` | INFO |

# Galadriel Harvester Configuration File
You can find the default Galadriel Harvester configuration file at `conf/harvester/harvester.conf`

| Configuration | Description | Required | Default
|--|--|--|--|
| `spire_socket_path` | SPIRE Server Socket of the instance to manage. | | /tmp/spire-server/private/api.sock |

> | `server_address` | Upstream Galadriel Server DNS name or IP address with port. | Yes | localhost:8085 |

Reviewer: nit: it has no default value, it fails if not set
```suggestion
| `server_address` | Upstream Galadriel Server DNS name or IP address with port. | Yes | |
```
